x11-packages/alacritty/build.sh

```diff
@@ -3,98 +3,79 @@ TERMUX_PKG_DESCRIPTION="A fast, cross-platform, OpenGL terminal emulator"
 TERMUX_PKG_LICENSE="Apache-2.0, MIT"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
 TERMUX_PKG_VERSION="0.15.1"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_REVISION=3
 TERMUX_PKG_SRCURL=https://github.com/alacritty/alacritty/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
 TERMUX_PKG_SHA256=b814e30c6271ae23158c66e0e2377c3600bb24041fa382a36e81be564eeb2e36
 TERMUX_PKG_DEPENDS="fontconfig, freetype, libxi, libxcursor, libxrandr"
 TERMUX_PKG_BUILD_DEPENDS="libxcb, libxkbcommon, ncurses"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
 
-__cargo_fetch_dep_source_for_rust_windowing() {
-	local _name="$1"
-	local _version
-	_version=$(cargo metadata --format-version=1 --no-deps | jq -r ".packages[0].dependencies[] | select(.name==\"$_name\") | .req")
-	_version="${_version/^/}"
-	local _url="https://github.com/rust-windowing/$_name/archive/refs/tags/v$_version.tar.gz"
-	local _path="$TERMUX_PKG_CACHEDIR/$_name-v$_version.tar.gz"
-	termux_download "$_url" "$_path" SKIP_CHECKSUM
-	tar xf "$_path" -C "$TERMUX_PKG_SRCDIR"
-	mv "$_name-$_version" "$_name-source"
-}
-
-termux_step_pre_configure() {
+termux_step_configure() {
 	termux_setup_cmake
 	termux_setup_rust
+	cargo clean
+	cargo vendor
 
-	: "${CARGO_HOME:=$HOME/.cargo}"
-	export CARGO_HOME
-
-	__cargo_fetch_dep_source_for_rust_windowing "winit"
-	__cargo_fetch_dep_source_for_rust_windowing "glutin"
-
-	patch="$TERMUX_PKG_BUILDER_DIR/patch-root-Cargo.diff"
-	patch -p1 -d "$TERMUX_PKG_SRCDIR" < "$patch"
-
-	cat "$TERMUX_PKG_BUILDER_DIR"/winit-*.diff | patch -p1 -d "$TERMUX_PKG_SRCDIR/winit-source"
-	cat "$TERMUX_PKG_BUILDER_DIR"/glutin-*.diff | patch -p1 -d "$TERMUX_PKG_SRCDIR/glutin-source"
-
-	cargo update
+	for dir in ./vendor/rustix*; do
+		[ -d "$dir" ] || continue
+		if [ -d "$dir/src/backend/libc/shm" ] && \
+			[ -d "$dir/src/backend/linux_raw/shm" ]; then
+			rm -rf "$dir/src/backend/libc/shm/"*
+			cp "$dir/src/backend/linux_raw/shm/"* \
+				"$dir/src/backend/libc/shm/"
```

Comment:
`[[` bash test `]]` is preferred here, and the line continuations are a bit excessive considering the rest of the build script's line lengths.

Also `-d "$dir/src/backend/libc/shm"` cannot be true if `$dir` doesn't exist,
so the `-d "$dir"` is superfluous.
```suggestion
		if [[ -d "$dir/src/backend/libc/shm" && -d "$dir/src/backend/linux_raw/shm" ]]; then
			rm -rf "$dir/src/backend/libc/shm/"*
			cp "$dir/src/backend/linux_raw/shm/"* "$dir/src/backend/libc/shm/"
```